### ISSUE TITLE
Search Improvement (Merge search into get_all & allow multiple search key/values)

### ISSUE THREAD

**Issue by kshahzada** [gtalarico/airtable-python-wrapper#122]
A minor inconvenience I have is:

- I'd like to be able to pass filter key and value into get_all as optional args - I've written wrappers for this in my code, but I think it's a convenience feature that could easily be added to the source
- I'd like to be able to pass an object of multiple search keys and values to get_all (or search) - I think this could either be an array of pairs of key/values or a dict

I'm happy to add this if this is something the maintainers are happy with

**Comment by gtalarico** [comment]
>  filter key and value into `get_all` as optional args?

Can you provide an example of what you mean? `.search` can be used for filtering.

> I'd like to be able to pass an object of multiple search keys and values to get_all (or search

I agree with this one. I would like to start a new major version soon and "searching/filtering"  is one of the areas I think should be re-worked.

eg. 

```python
table.search(name="User", age=18) # return matches of `name` AND `age`
```
(To support OR, we would need to create a more complex query building)

In the meantime, we can achieve multiple param filtering by  building the formula pretty easily - 
the api above would actually just do this for you:

```python
>>> table = Airtable("appu15p8579gOPqm7", "test table", api_key=os.environ["AIRTABLE_API_KEY"])

>>> def and_query(**params: dict) -> str:
    ... params_str = ", ".join([f"{k}={repr(v)}" for k, v in params.items()])
    ... return f"AND({params_str})"

>>> formula = and_query(**{"Name": "A", "Age": 20})
>>> results = table.get_all(formula=formula)
# same as 
# table.get_all(formula='AND(Name="A", Age="A")')

>>> print(formula)
AND(Name='A', Age=20)

>>> print(results)
[{'id': 'recP6SdMCGZOGyTTg', 'fields': {'Name': 'A', 'test': 1, 'Age': 20}, 'createdTime': '2021-07-04T18:57:46.000Z'}]
```

![image](https://user-images.githubusercontent.com/9513968/124397328-13dec780-dcc4-11eb-9b40-6a0af2552fa2.png)

**Comment by gtalarico** [comment]
Something like this is being implemented in #126 
Wip docs
https://airtable-python-wrapper.readthedocs.io/en/version-2/api.html#pyairtable.formulas.match
